### PR TITLE
get travis build passing

### PR DESCRIPTION
**Pull request by pat-s**
did some adjustments and timing out in my fork; let's see if the PR build works :smile: 

(the branch name is not correct and can be ignored)

EDIT: Appears to be standard travis timout limit (~ 50 min). I thought you have increased build time here @dakcarto ?
build log: https://travis-ci.org/qgis/homebrew-qgisdev/builds/305831818?utm_source=github_status&utm_medium=notification

### CODE REVIEW

**Comment by pat-s** [comment]
@dakcarto did you have time to take a look at this? :smiley:

File: .travis.yml [hunk 1]
```diff
@@ -12,9 +12,9 @@ before_install:
       echo "Skipping CI; no changed formulae found in $TRAVIS_BUILD_DIR";
       travis_terminate 0;
     else
-      echo "Changed formulae are ${CHANGED_FORMULAE}";
-    fi
+      echo "Changed formulae are ${CHANGED_FORMULAE}"; fi
```
Comment (dakcarto) [inline .travis.yml:20]: What's the point of this?

File: .travis.yml [hunk 2]
```diff
@@ -12,9 +12,9 @@ before_install:
...
   - export HOMEBREW_REPOSITORY="$(brew --repo)"
+  - /usr/bin/yes | pip2 uninstall numpy
```
Comment (dakcarto) [inline .travis.yml]: Why would `numpy` installed into Python **2** affect this?

File: travis/script.sh [hunk 3]
```diff
@@ -19,5 +19,16 @@ set -e
...
+  # add progress to ensure Travis doesn't complain about no output
+  while true; do
+    sleep 30
+    if jobs -rp | grep ${PID} >/dev/null; then
+      echo "."
+    else
+      echo
+      break
+    fi
+  done
```
Comment (dakcarto) [inline travis/script.sh:33]: This makes sense. Nice catch.

**Review by dakcarto — CHANGES_REQUESTED** [review]
To get build times down, we will need to revamp the formula to utilize `ccache` and implement the tricks @m-kuhn has done recently with caching to get the QGIS main code repo building faster.

Nonetheless, I doubt we will be able to compile it and run unit tests before timing out, at least on Travis CI's limited resource (e.g. CPUs) macOS images.

Apparently, the default timeout of ~ 1 hour is still in effect on this repo, or was reverted. Or, maybe I'm confused and this repo was never considered in the conversation with them.

Travis is awesome as a free service, though often a PITA for QGIS compilation needs.

File: .travis.yml (hunk 2 above)
Comment (dakcarto) [reply .travis.yml]: Also, `pip uninstall -y numpy || true` would be the shortened command (ignoring version). See `pip uninstall -h`.

File: .travis.yml (hunk 1 above)
Comment (pat-s) [reply .travis.yml:20]: if you mean the linebreak, that's by accident

File: .travis.yml (hunk 2 above)
Comment (pat-s) [reply .travis.yml]: Its installed by default and interferes with any `numpy` installation from brew. See the last build of `master`.

Comment (pat-s) [reply .travis.yml]: > Also, pip uninstall -y numpy || true would be the shortened command (ignoring version)

You need to use `pip2` here as `pip` without a version is not known for some reason when using `xcode8.3`.

**Comment by pat-s** [comment]
> Nonetheless, I doubt we will be able to compile it and run unit tests before timing out, at least on Travis CI's limited resource (e.g. CPUs) macOS images.

We could try using build stages on Travis. We do something similar on [here](https://github.com/mlr-org/mlr/blob/master/.travis.yml).

> Apparently, the default timeout of ~ 1 hour is still in effect on this repo, or was reverted. Or, maybe I'm confused and this repo was never considered in the conversation with them.

It seems to me that its 50 min according to my last build and this looks like the default for travis if I am not mistaken :smile: 

> To get build times down, we will need to revamp the formula to utilize ccache and implement the tricks @m-kuhn has done recently with caching to get the QGIS main code repo building faster.
Ok, I'll take a look

File: .travis.yml (hunk 2 above)
Comment (dakcarto) [reply .travis.yml]: > You need to use pip2 here as pip without a version is not known for some reason when using xcode8.3.

That's what "(ignoring version)" means.

**Pull request closed without merging**